prql-compiler/src/sql/codegen.rs

```diff
@@ -549,28 +550,22 @@ fn is_keyword(ident: &str) -> bool {
 }
 
 pub(super) fn translate_ident_part(ident: String, ctx: &Context) -> sql_ast::Ident {
+    // We'll remove this when we get the new dbt plugin working (so no need to
+    // integrate into the regex)
     let is_jinja = ident.starts_with("{{") && ident.ends_with("}}");
-
-    // TODO: can probably represent these with a single regex
-    fn starting_forbidden(c: char) -> bool {
-        !(('a'..='z').contains(&c) || matches!(c, '_' | '$'))
-    }
-    fn subsequent_forbidden(c: char) -> bool {
-        !(('a'..='z').contains(&c) || ('0'..='9').contains(&c) || matches!(c, '_' | '$'))
+    lazy_static! {
+        // One of:
+        // - `*`
+        // - An ident starting with `a-z_\$` and containing other characters `a-z0-9_\$`
+        static ref VALID_BARE_IDENT: Regex = Regex::new(r"^((\*)|(^[a-z_\$][a-z0-9_\$]*))$").unwrap();
```

Comment:
I'm actually not sure this is simpler! But it is consolidated. And it's much faster, though probably we shouldn't weigh that too highly.

Reply:
Actually it's not too bad, complexity wise. And now that I think about performance - before it was much worse.

If you want to make it more readable, we could replace the regex string with call to [pomsky_macro](https://docs.rs/pomsky-macro/0.7.0/pomsky_macro/) ([docs](https://pomsky-lang.org/docs/get-started/quick-start/)).

Equivalent pomsky expression would be (I hope at least):
```
^ ('*' | [ascii_lower '_$'] [ascii_lower ascii_digit '_$']* ) $
```

Reply:
Oh yes I'm up for that! I've been following the project albeit from a distance. 

I won't do it quite yet — requires adding a dependency — but v open to doing it in the future